src/csharp/Grpc.Core/GrpcEnvironment.cs

```diff
@@ -249,6 +274,7 @@ public static void SetHandlerInlining(bool inlineHandlers)
         private GrpcEnvironment()
         {
             GrpcNativeInit();
+            batchContextPool = new DefaultObjectPool<BatchContextSafeHandle>(() => BatchContextSafeHandle.Create(this.batchContextPool), batchContextPoolSharedCapacity, batchContextPoolThreadLocalCapacity);
```

Comment:
optional: since `DefaultObjectPool` is only used for batch contexts, simplify by replacing injection of the batch context factory lambda with an instance method on `DefaultObjectPool`, and also use only `BatchContextSafeHandle` objects in the tests?

Reply:
the default object pool will be used for RequestCallContextSafehandle objects in a followup PR (for same reasons).